src/components/calcite-date-picker/utils.ts

```diff
@@ -42,6 +42,9 @@ function getSupportedLocale(lang = "") {
 
   if (lang.includes("-")) {
     lang = lang.replace(/(\w+)-(\w+)/, (_match, language, region) => `${language}-${region.toUpperCase()}`);
+    if (!locales.includes(lang)) {
+      return locales.includes(lang.split("-")[0]) ? lang.split("-")[0] : "en";
```

Comment:
Can you save `lang.split("-")[0])` as a const so that you don't have to split it twice here? Otherwise looks good! 👍